src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts

```diff
@@ -0,0 +1,421 @@
+                `
+              )}
+            </paper-listbox>
+          </paper-dropdown-menu>
+        </div>
+      `;
+    }
+
+    return html`${labelAndDescription}
+      <p>${item.value}</p>`;
+  }
+
+  private _isEnumeratedBool(item): boolean {
+    // Some Z-Wave config values use a states list with two options where index 0 = Disabled and 1 = Enabled
+    // We want those to be considered boolean and show a toggle switch
+    const disabledStates = ["disable", "disabled"];
+    const enabledStates = ["enable", "enabled"];
+
+    if (item.configuration_value_type !== "enumerated") {
+      return false;
+    }
+    if (!("states" in item.metadata)) {
+      return false;
+    }
+    if (!(0 in item.metadata.states) || !(1 in item.metadata.states)) {
+      return false;
+    }
+    if (
+      disabledStates.includes(item.metadata.states[0].toLowerCase()) &&
+      enabledStates.includes(item.metadata.states[1].toLowerCase())
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  private _switchToggled(ev) {
+    this._updateConfigParameter(ev.target, ev.target.checked ? 1 : 0);
+  }
+
+  private _dropdownSelected(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === ev.target.selected) {
+      return;
+    }
+
+    this._updateConfigParameter(ev.target, parseInt(ev.target.selected));
+  }
+
+  private debouncedUpdate = debounce((target) => {
+    const value = parseInt(target.value);
+    this._config![target.key].value = value;
+
+    this._updateConfigParameter(target, value);
+  }, 1000);
+
+  private _numericInputChanged(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === parseInt(ev.target.value)) {
+      return;
+    }
+    this.debouncedUpdate(ev.target);
+  }
+
+  private _updateConfigParameter(target, value) {
+    setNodeConfigParameter(
+      this.hass,
+      this.configEntryId!,
+      this._nodeId!,
+      target.property,
+      value,
+      target.propertyKey ? target.propertyKey : undefined
+    );
+    this._config![target.key].value = value;
+  }
+
+  private get _device(): DeviceRegistryEntry | undefined {
+    return this._deviceRegistryEntries
+      ? getDevice(this._deviceRegistryEntries, this.deviceId)
+      : undefined;
+  }
+
+  private get _nodeId(): number | undefined {
+    const identifier = getIdentifier(this._device!);
+
+    if (!identifier) {
+      this._error = "couldnt_get_node_id_from_device";
+      return undefined;
+    }
+
+    return parseInt(identifier[1].split("-")[1]);
+  }
+
+  private async _fetchData() {
+    if (!this.configEntryId || !this._deviceRegistryEntries) {
+      return;
+    }
+
+    if (this._device === undefined || this._nodeId === undefined) {
+      this._error = "device_not_found";
+      return;
+    }
+
```

Comment:
const nodeId = this._nodeId;
    if (!nodeId) {
      this._error = "couldnt_get_node_id_from_device";
      return;
}

Reply:
I never really liked that error after I added it, so I just got rid of it, if we can't get the node ID then effectively the device is not found 🙂